sgt-back-end/index.js

```diff
@@ -0,0 +1,189 @@
+const express = require('express');
+const app = express();
+const pg = require('pg');
+
+app.use(express.json());
+
+// only create ONE pool for your whole server
+const db = new pg.Pool({
+  connectionString: 'postgres://dev:dev@localhost/studentGradeTable',
+  ssl: {
+    rejectUnauthorized: false
+  }
+});
+
+// GET / api / grades returns all rows from the "grades" table. The client should receive an array of objects.If there happens to be no rows, an empty array is ok. The result could be a 200 or a 500.
+
+/** http GET localhost:3000/api/grades */
+
+app.get('/api/grades', (req, res) => {
+  const sql = `
+    select *
+      from "grades"
+      `;
+  db.query(sql)
+    .then(result => {
+      res.status(200).json(result);
+    }
+    )
+    .catch(err => {
+      console.error(err);
+      res.status(500).json({
+        error: 'An unexpected error occurred.'
+      });
+    });
+});
+
+// POST / api / grades inserts a new grade into the "grades" table and returns the created grade.The client should receive an object, not an array.The result could be a 201, 400, or 500.
+
+/** http POST localhost:3000/api/grades course=courseName name='Namey Name' score=# */
+
+app.post('/api/grades', (req, res) => {
+  const content = req.body;
+  if (typeof content === 'undefined') {
+    res.status(400).json({
+      error: 'content is a required field'
+    });
+    return;
+  }
+  content.score = parseInt(content.score);
+  if (typeof content.score !== 'number' ||
+      content.score < 0 ||
+      content.score > 100 ||
+      typeof content.name === 'undefined' ||
+      typeof content.course === 'undefined') {
+    res.status(400).json({
+      error: 'content must include valid course, name, and score'
+    });
+  }
```

Comment:
This should be 2 if statements so the error messages can be more specific, also you can simplify it a bit by not using `typeof`.

```JAVASCRIPT
if (!content.name, !content.course, !content.score) {
  res.status(400).json({error: 'content must include valid course, name, and score'});
  return;
}
if (content.score < 0 || content.score > 100) {
  res.status(400).json({error: 'Score must be an integer between 0 and 100'});
  return;
}
```

Reply:
cool, on it!